module/zfs/sha256.c

```diff
@@ -119,9 +120,25 @@ zio_checksum_SHA256(const void *buf, uint64_t size, zio_cksum_t *zcp)
 	for (i = 0; i < padsize; i += 64)
 		SHA256Transform(H, pad + i);
 
-	ZIO_SET_CHECKSUM(zcp,
-	    (uint64_t)H[0] << 32 | H[1],
-	    (uint64_t)H[2] << 32 | H[3],
-	    (uint64_t)H[4] << 32 | H[5],
-	    (uint64_t)H[6] << 32 | H[7]);
+	zcp->zc_word[0] = (uint64_t)H[0] << 32 | H[1];
+	zcp->zc_word[1] = (uint64_t)H[2] << 32 | H[3];
+	if (!truncate) {
+		zcp->zc_word[2] = (uint64_t)H[4] << 32 | H[5];
+		zcp->zc_word[3] = (uint64_t)H[6] << 32 | H[7];
+	}
+}
+
+void
+zio_checksum_SHA256(const void *buf, uint64_t size, zio_cksum_t *zcp)
+{
+	zio_checksum_SHA256_common(buf, size, zcp, B_FALSE);
+}
+
+/*
+ * Same as zio_checksum_SHA256, but truncates to allow 96 bits for MAC
```

Comment:
Yes it should. An older implementation used only 96 bits for the MAC. I will change it now and push it when I have my local branch cleaned up a little bit.